dist/knockout-pre-rendered.js

```diff
@@ -582,134 +587,215 @@
-    // or the field parameter doesn't, we try the text and value binding
-    var fieldAccessor =
-      !value && "field" in initPropertyWriters
-        ? initPropertyWriters["field"]
-        : ko.isObservable(value)
-        ? value
-        : isPlainObject(value) && "field" in value
-        ? ko.isObservable(value["field"])
-          ? value["field"]
-          : initPropertyWriters["field"]
-        : undefined;
+    if (ko.isObservable(fieldAccessor)) {
+      fieldAccessor(value, ko.unwrap(fieldAccessor));
+    } else {
+      var propertyWriter = getPropertyWriter(
+        allBindings,
+        propertyWritersBindingName,
+        key
+      );
+      if (propertyWriter) {
+        propertyWriter(value);
+      }
+    }
+  }
 
-    if (!fieldAccessor) {
-      var supportedBindings = [
-        "text",
-        "textInput",
-        "value",
-        "checked",
-        "html",
-        "visible",
-        "enable",
-        "disable"
-      ];
-      for (var i = 0; i < supportedBindings.length && !fieldAccessor; i++) {
-        var bindingName = supportedBindings[i];
-        var accessor = allBindings.get(bindingName);
-        if (accessor && ko.isObservable(accessor)) {
-          fieldAccessor = accessor;
-        } else if (bindingName in propertyWriters) {
-          fieldAccessor = propertyWriters[bindingName];
-        }
+  function getPropertyWriter(allBindings, propertyWritersBindingName, key) {
+    if (propertyWritersBindingName) {
+      var propertyWriters = allBindings.get(propertyWritersBindingName);
+      if (propertyWriters) {
+        return propertyWriters[key];
       }
     }
+    return null;
+  }
 
-    // Finally, set the field value.
-    if (fieldAccessor) {
-      fieldAccessor(fieldValue, unwrappedValue);
+  // Returns true if the init binding specifies a field in which to write values extracted from the markup.
+  function hasFieldOverride(initBinding) {
+    if (isPlainObject(initBinding)) {
+      // init: { field: fieldName }
+      if (initBinding.hasOwnProperty("field")) {
+        return true;
+      }
+    } else if (initBinding !== true) {
+      // init: fieldName
+      return true;
     }
+    return false;
   }
 
   // This binding handler initializes an observable to a value from the HTML element
   ko.bindingHandlers.init = {
     init: function(element, valueAccessor, allBindings, viewModel) {
-      var value = valueAccessor();
+      var initBinding = valueAccessor();
 
-      if (isObjectWithExplicitValues(value)) {
-        setExplicitObjectValues(value, viewModel, allBindings);
-      } else if (hasAttributeBinding(allBindings)) {
-        initAttributeObservables(element, value, allBindings);
-      } else {
-        initObservable(element, value, allBindings);
+      if (isObjectWithExplicitValues(initBinding)) {
+        setExplicitObjectValues(initBinding, viewModel, allBindings);
+      }
+
+      if (initBinding !== true) {
```

Comment:
Just a suggestion: you could consider inverting the if-statement and doing an early return. This both reduces the nesting and allows the boolean expression to be positive, instead of a negation (which is slightly harder to read).

Reply:
Agreed, that would make it cleaner. In fact, I'll move the `if (initBinding === true)` check to the very top, with an early return. This will also bypass the `isPlainObject` test.